text/0000-hybrid-tx-building.md

```diff
@@ -0,0 +1,281 @@
+
+- Title: `hybrid-tx-building`
+- Authors: [@lehnberg](mailto:daniel.lehnberg@protonmail.com)
+- Start date: `Mar 19, 2020`
+- RFC PR: Edit if merged: [mimblewimble/grin-rfcs#0000](https://github.com/mimblewimble/grin-rfcs/pull/0000) 
+- Tracking issue: [Edit if merged with link to tracking github issue]
+
+---
+
+# Summary
+[summary]: #summary
+
+This RFC establishes a baseline for transaction building in Grin between users and services, through the introduction of a new 'hybrid' method. Transactions are initiated used two standardized formats, **QR code** and **blob**, and can then be completed over Tor or clearnet using https.
+
+To support this, the transaction slate data at each step in the process is redesigned and minified.
+
+The introduction of this method improves security as it does not support unsafe communication methods, and it improves usability as friction and potential drop-off points are reduced in the transaction building flow.
```

Comment:
I don't think this improves security much at all. Unless we're adding e2e encryption, security will be minimally impacted.

Reply:
The hybrid method enforces https so communication would be secured over TLS/SSL. This improves security over plain http.

As discussed in Keybase, the hybrid method can support e2e encryption, but it does add more friction as key or password exchange is then required. I think it would be a worth while optional feature to have, but I kept it out of the scope of this RFC. Similarly to the matter of https deprecation, encrypted slates is an independent matter - it can be added with or without the hybrid method.

Reply:
"That being said, since the proposed approach would support Tor-only as well, the matter of whether or not to deprecate https is independent of what is being proposed here, and should probably be handled separately"

How is depreciating http in scope, but deprecating https is not?

Reply:
@DavidBurkett I'm putting together an RFC for deprecating both http and https. I think I want to advocate for using armored slates and being as transport agnostic as possible in the core wallet, only supporting methods that have acceptable amounts of security and privacy (which leaves importing files/slates directly and Tor). I'll ping you when I get the outline together if you are interested, would like your input in some places before I go too far.

Reply:
> How is depreciating http in scope, but deprecating https is not?

I'm probably unclear in my writing. Can you point me to the passage that suggested http will be deprecated as part of this RFC, so I can correct it?